### PR TITLE
Enable sbt-dynver

### PR DESCRIPTION
**Pull request by renatocaval**
Targeting 2.8.x with the intention to merge back in `master`.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -86,6 +86,39 @@ libraryDependencies in ThisBuild ++= Seq(
...
+    throw new MessageOnlyException(
+      s"Failed to derive version from git tags. Maybe run `git fetch --unshallow`? Version: $v"
+    )
+  s
+}
+
+// this is overrides interplay release settings with some adaptations for working with sbt-dynver
+// moreover, it contain only bits necessary for play-json, nothing else
+lazy val releaseSettings: Seq[Setting[_]] = Seq(
+  releasePublishArtifactsAction := PgpKeys.publishSigned.value,
```
Comment (marcospereira) [inline build.sbt]: I thought that it was only necessary to override `releaseProcess`. Are the others really necessary here?

File: build.sbt [hunk 2]
```diff
@@ -86,6 +86,39 @@ libraryDependencies in ThisBuild ++= Seq(
...
+Global / onLoad := (Global / onLoad).value.andThen { s =>
+  val v = version.value
+  if (dynverGitDescribeOutput.value.hasNoTags)
+    throw new MessageOnlyException(
+      s"Failed to derive version from git tags. Maybe run `git fetch --unshallow`? Version: $v"
+    )
+  s
+}
+
+// this is overrides interplay release settings with some adaptations for working with sbt-dynver
```
Comment (marcospereira) [inline build.sbt]:
```suggestion
// this overrides interplay release settings with some adaptations for working with sbt-dynver
```

File: build.sbt (hunk 1 above)
Comment (renatocaval) [reply build.sbt]: It's true. We only need to overwrite `releaseProcess`. 

The original [code](https://github.com/playframework/interplay/blob/master/src/main/scala/interplay/PlayReleaseBase.scala#L40-L61) in interplay had some embedded functions that were used to define the process. 

I started by copying everything and then while cleaning up, I remove all of them. It's not needed any more, we can fall back to only re-defining the `releaseProcess`.

Comment (renatocaval) [reply build.sbt]: @marcospereira, this is updated now

File: build.sbt [hunk 3]
```diff
@@ -127,6 +141,22 @@ lazy val root = project
...
   .settings(commonSettings)
+  .settings(
+    Seq(
+      // this overrides releaseProcess to make it work with sbt-dynver
+      releaseProcess := {
+        import ReleaseTransformations._
+        Seq[ReleaseStep](
+          checkSnapshotDependencies,
+          runClean,
+          releaseStepCommandAndRemaining("+test"),
```
Comment (ignasi35) [inline build.sbt]:
```suggestion
          releaseStepCommandAndRemaining("+test"), // <- this needs to be removed when releasing from Travis
```

😂

**Pull request merged**